### ISSUE TITLE
Add more benchmarks.

### ISSUE THREAD

**Issue by nsthorat** [tensorflow/tfjs#77]
_From @nsthorat on October 19, 2017 14:54_

We have several benchmarks here:
https://deeplearnjs.org/demos/benchmarks/

We would love to see benchmarks for more `NDArrayMath` functions.

New benchmarks should be modeled after this file:
https://github.com/PAIR-code/deeplearnjs/blob/master/demos/benchmarks/reduction_ops_benchmark.ts

Reduction ops are special as they are grouped, but new benchmarks for math functions should be put in their own file, with only one option (instead of having switches), unless operations can be logically grouped.

_Copied from original issue: tensorflow/tfjs-core#233_

**Comment by nsthorat** [comment]
_From @easadler on October 26, 2017 15:1_

@nsthorat I love what you guys are doing. Would love to try and help out. Any `NDArrayMath` benchmarks in particular would you like to see?

**Comment by nsthorat** [comment]
@easadler Thanks Evan!

A nice place to start might be batchNormalization3D.

**Comment by nsthorat** [comment]
_From @easadler on October 28, 2017 17:26_

Awesome, got started this morning.

**Comment by nsthorat** [comment]
_From @erikwilson on November 25, 2017 18:54_

When I look at the benchmarks for matmul CPU vs GPU in Chrome it appears that using the GPU is always the better choice. However when I am using Electron with the matmul intro training example the CPU is about 10x faster than the GPU. Is there something special about meanSquaredCost or training that might cause this?

**Comment by nsthorat** [comment]
_From @erikwilson on November 25, 2017 19:31_

Playing some with intro training, if I don't await the cost value and math.scope I can get the CPU to be only ~2x as fast as the GPU. I am wondering if there is any I/O that these benchmarks should include?

**Comment by nsthorat** [comment]
Closing this since it's outdated.
